src/states/common/base.rs

```diff
@@ -311,16 +311,19 @@ pub trait Base: Display {
         conn_info: ConnectionInfo,
         msg: Bytes,
         token: Token,
-        _outbox: &mut dyn EventBox,
+        outbox: &mut dyn EventBox,
     ) -> Transition {
         let log_ident = LogIdent::new(self);
-        self.network_service_mut().send_message_to_next_target(
+        if self.network_service_mut().send_message_to_next_target(
             msg,
             token,
             conn_info.peer_addr,
             log_ident,
-        );
-        Transition::Stay
+        ) {
+            self.handle_peer_lost(conn_info.peer_addr, outbox)
```

Comment:
What if that message is to a client (from a vault)?

Reply:
`handle_unsent_message` is called only for nodes.

Reply:
Yes, indeed. wonder if that is an issue for vault messages (i.e does the message still end up in that cache).

Reply:
You mean message sent from a vault to another vault(s)? Such messages go through routing normally which I think is what we want.

Reply:
Well I mean both, but i think in this case I was more thinking vault -> client. Routing still have the quic_p2p instance so I think sending message to client also go through it?

Reply:
Yes, but you need to use `Peer::Client` when sending to clients and that's what you'll get back in `UnsentUserMessage` and `SentUserMessage`, so we always know whether the event is for client or node and only handle the ones for node.

Reply:
But do we do the same check when sending a message and not adding it to the cache?

Reply:
There is a separate function `send_message_to_client` for that and we don't insert it to the cache there.

Reply:
ok make sense.